intermediate-js.md

```diff
@@ -1,39 +1,73 @@
+
 4. Why is testing important?
 
+  Testing is important for a number of reasons. Users and investors gain trust in companies who test their code. Testing helps developers check small blocks of code to make sure they work. Testing may also help the developer understand what is needed to code in the future to add new/necessary functionality.
+
 5. What is a higher order function?
 
+  A function that takes in or outputs another function.
+
 6. What is the difference between a class and an object?
 
+  An object in javascrpt has key and value pairs as well as behavior. Classes in javascript used to construct objects.
+
 7. What did you learn during the group project this week? Please include any additional feedback you may have.
 
+  Battleship was a huge step for me this week. I am starting to understand the full potential for React. Working in a group for this project was difficult because there were so many ways of implementing our ideas. At times it was frustrating but overall a great experience.
+
 
 ### HTML/CSS Review questions: First, try to answer each question on your own then Google the answer to further your knowledge.
 
 1. How do you link a CSS file to your HTML page?
 
+  <script src="file_path_goes_here.css" />
+
 2. What is the difference between a div and a span?
 
+  A div is used to nest html elements. Span is used to group elements in order to style or perform js on them.
+
 3. What is a CSS class? When should you use an id instead of a class?
 
+  CSS classes are ways of organizing css blocks and applying that specific code to html elements. Classes may be used on multiple elements while id is unique and can only be applied to one element.
+
 4. Name 4 semantic HTML tags.
 
+  header tags: <header>
+  footer tag: <footer>
+  nav tag: <nav>
+  main tag: <main>
+
 5. What are three options for creating responsive design?
+  Responsive design is used to give the user an experience that responds to the device the user is on.
+  flex
+  scale
 
 
 ### Stretch: The following questions are potential interview questions. First, try to answer each question on your own then Google the answer to further your knowledge.
 
 1. What is front end development? Can you identify any tools/skills that are uniquely required of front end developers?
 
+Front end development is building the part of the app that the user interacts with. UX and UI are used to ensure the user has the best possible experience. For web based apps, HTML is a must in order to code the front end.
+
 2. What is block scope in JavaScript?
 
+  Block scope has to do with how variables are read in JavaScript. Within certains scopes, such as a function, JavaScript will not read specific variables. Using let, var, and const relate to block scope.
```

Comment:
Very close! Let and const assignments actually utilize block scope. Var only enforces function scope. You can think of blocks as code that is between curly braces but not as part a of a function (conditional blocks/loops).